multiboot2/src/module.rs

```diff
@@ -48,12 +44,21 @@ impl ModuleTag {
     }
 }
 
+impl crate::TagTrait for ModuleTag {
+    fn dst_size(base_tag: &Tag) -> usize {
+        // The size of the sized portion of the module tag.
+        let tag_base_size = 16;
+        base_tag.size as usize - tag_base_size
```

Comment:
protect against invalid tag_base.size() values. panic if tag_base_size is bigger than tag_base_size